lib/reek/examiner.rb

```diff
@@ -44,28 +42,32 @@ def initialize(source,
                    configuration: Configuration::AppConfiguration.default,
                    smell_repository_class: Smells::SmellRepository)
       @source           = Source::SourceCode.from(source)
-      @collector        = CLI::WarningCollector.new
```

Comment:
I quickly checked and this seems to be the only place where we use WarningCollector, this means the whole class can be removed as well.